lib/redis/namespace.rb

```diff
@@ -364,6 +364,20 @@ def call_with_namespace(command, *args, &block)
 
       (before, after) = handling
 
+      # Avoid modifying the caller's (pass-by-reference) arguments.
+      def clone_args(arg)
+        if arg.is_a?(Array)
+          arg.map {|sub_arg| clone_args(sub_arg)}
+        elsif arg.is_a?(Hash)
+          arg.map {|k, v| [clone_args(k), clone_args(v)]}.to_h
```

Comment:
Looks like the tests are failing because Array#to_h isn't available in older versions of ruby that redis-namespace supports.